src/lib.rs

```diff
@@ -65,12 +67,54 @@ impl OfferHandler {
     }
 
     /// Adds an offer to be paid with the amount specified. May only be called once for a single offer.
-    pub fn pay_offer(&mut self, _offer: Offer, _amount: u64) -> Result<(), OfferError> {
-        /*
-           Check if we've already added offer -> error.
-           Add offer to state map.
-           Create invoice request and push to offer queue.
-        */
+    pub async fn pay_offer(
+        &self,
+        offer: Offer,
+        amount: Option<u64>,
+        network: Network,
+        client: Client,
+        blinded_path: BlindedPath,
+    ) -> Result<(), OfferError<Secp256k1Error>> {
+        sleep(Duration::from_secs(10)).await;
+
+        validate_amount(&offer, amount).await?;
+
+        // For now we connect directly to the introduction node of the blinded path so we don't need any
+        // intermediate nodes here. In the future we'll query for a full path to the introduction node for
+        // better sender privacy.
+        connect_to_peer(
+            client.clone(),
+            blinded_path.introduction_node_id,
+            String::from(""),
```

Comment:
The empty String is just a placeholder for now!

We had talked about connecting directly to the introduction node peer here if we aren't already connected, but there's one missing piece... the Blinded Path returned by the offer only provides the node's public key, not the address. I'll ask LDK if they'd be opposed to adding that info to the Blinded Path struct.

Though if that doesn't work out for whatever reason... we can try something like... using lnd's GetNodeInfo to grab the address information.

Reply:
Ok, I talked to Jeff from LDK and he suggested using this PR which is in LDK 0.0.119: https://github.com/lightningdevkit/rust-lightning/pull/2723
We can wait on the MessageRouter to return a ConnectionNeeded event for us to handle if it can't find a route. The big change we'd need here is to manage another NetworkGraph... Since we already have that on LND's end maybe that's not ideal. 

But since the MessageRouter provides a chance at doing a multi-hop message send to the blinded route (if there's actually a route supporting onion messages available), I'm wondering if that'd be a good way to go? On LND's end, we can only make payments with SendToRoute, not send the invoice request, right? Thoughts?

Reply:
I've been thinking more about this. Implementing the full MessageRouter doesn't really make sense for us right now for the CLI pay_offer command, since we start up a new instance of LndkOnionMessenger. In this scenario, I assume we don't have time to sync a whole new network graph to get MessageRouter's benefits.

Eventually though it will make sense to have when we have a grpc server connecting users to an LNDK instance which is already synced. The MessageRouter will help us find multi-hop routes for routing to an introduction node, etc

So I'll just use the describe_graph approach for now.

I would love a sanity check on this thinking however. :)